src/main/java/io/confluent/idesidecar/restapi/proxy/ClusterProxyRequestProcessor.java

```diff
@@ -50,10 +50,13 @@ public Future<ClusterProxyContext> process(ClusterProxyContext context) {
       // If Kafka, send the request to the Kafka REST server
       // using a generic HTTP client
       case KAFKA -> proxyHttpClient.send(context);
-      // If Schema Registry, use the cached SchemaRegistryClient instance
-      // to send the request to the Schema Registry server. Constructing the full
-      // HTTP request ourselves is error-prone and unnecessary.
-      case SCHEMA_REGISTRY -> processSchemaRegistry(context);
+      case SCHEMA_REGISTRY ->
+          switch (context.getConnectionState().getType()) {
+            // For CCloud connections, use Proxy HTTP client
+            case CCLOUD -> proxyHttpClient.send(context);
+            // For all other connections, use SR client
+            case DIRECT, LOCAL, PLATFORM -> processSchemaRegistry(context);
```

Comment:
Remind me why users won't possibly have similar SSL issues for DIRECT or PLATFORM schema registries? Because the direct connections will have already been SSL tuned accordingly?

Reply:
Yes, it only applied to CCloud.

For the CCloud connection, we allow users to configure custom SSL certs via the extension settings. They're picked up by an [application-wide singleton instance of the WebClient](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/util/WebClientFactory.java#L75). We make sure that [updating the extension settings will update the WebClient's configuration](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/util/WebClientFactory.java#L89).

Direct/Platform connections rely on connection-specific SSL settings, which were correctly picked up.